src/FakeItEasy/TaskHelper.cs

```diff
@@ -1,14 +1,59 @@
 namespace FakeItEasy
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Reflection;
     using System.Threading.Tasks;
 
     internal static class TaskHelper
     {
+        private static readonly Lazy<Task> CachedCancelledTask = new Lazy<Task>(CreateGenericCancelledTask<int>);

```

Comment:
> Why int? Just because we need something and int is innocuous?

Yes.

> Could we instead use (or add) the value at `CachedCancelledTasks[typeof(int)]`?

Good point. Yes, we can totally do that.